wordpress-mobile/WordPress-Android
# FluxC: Delete site integration

Closes #5006. This PR implements the delete site flow that's introduced to FluxC in https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/266. Currently the only issue is that once a site is deleted the Site Picker doesn't refresh and still shows the deleted site in the list. I opened a separate issue for that, since it'll most likely happen for removed sites as well #5018.

The deleted site should be removed from DB [here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/develop/fluxc/src/main/java/org/wordpress/android/fluxc/store/SiteStore.java#L682), but due to the issue explained above, I am still not sure if that's the case or not.

To test:
* Create a new WordPress.com site
* Go into the settings of that site
* Delete the site
* You should be taken back to the Site Picker page
